cpp/arcticdb/async/async_store.hpp

```diff
@@ -367,67 +397,104 @@ class AsyncStore : public Store {
+        const std::shared_ptr<DeDupMap> &de_dup_map,
+        const BatchWriteArgs &args) override {
+        std::size_t write_count =
+            args.lib_write_count == 0 ? 16ULL : args.lib_write_count;
 
-        std::shared_ptr<std::vector<VariantKey>> res = std::make_shared<std::vector<VariantKey>>(key_segments.size());
+        std::shared_ptr<std::vector<VariantKey>> res =
+            std::make_shared<std::vector<VariantKey>>(key_segments.size());
 
         auto count = 0ULL;
-        for (std::size_t start = 0ULL, nxt; start < key_segments.size(); start = nxt) {
+        for (std::size_t start = 0ULL, nxt; start < key_segments.size();
+             start = nxt) {
             nxt = std::min(start + write_count, key_segments.size());
-            auto range = folly::Range(key_segments.begin() + start, key_segments.begin() + nxt);
-            auto key_segs = std::make_shared<std::vector<storage::KeySegmentPair>>();
+            auto range = folly::Range(key_segments.begin() + start,
+                                      key_segments.begin() + nxt);
+            auto key_segs =
+                std::make_shared<std::vector<storage::KeySegmentPair>>();
             auto key_seg_mutex = std::make_shared<std::mutex>();
             std::vector<folly::Future<folly::Unit>> futs;
             futs.reserve(range.size());
 
-            for(auto& kv : range) {
-                futs.emplace_back(async::submit_cpu_task(
-                    EncodeAtomTask(std::move(kv.first), ClockType::nanos_since_epoch(), std::move(kv.second), library_, codec_, count++))
-                                       .thenValue([de_dup_map, res](auto &&key_seg) {
-                    auto de_dup_key = de_dup_map ? de_dup_map->get_key_if_present(key_seg.atom_key()) : std::nullopt;
-
-                    if (de_dup_key) {
-                        ARCTICDB_DEBUG(log::version(), "Found existing key with same contents: using existing object {}", de_dup_key.value());
-                        (*res)[key_seg.id()] = (de_dup_key.value());
-                        return std::make_pair(storage::KeySegmentPair{}, false);
-                    } else {
-                        ARCTICDB_DEBUG(log::version(), "No existing key with same contents: writing new object {}", key_seg.atom_key());
-                        (*res)[key_seg.id()] = (key_seg.atom_key());
-                        return std::make_pair(std::forward<storage::KeySegmentPair>(key_seg), true);
-                    }
-                }).thenValue([key_segs=key_segs, key_seg_mutex=key_seg_mutex](auto && pair_key_write) {
-                        if (pair_key_write.second) {
-                            std::lock_guard kl{*key_seg_mutex};
-                            ARCTICDB_DEBUG(log::version(), "Enqueuing {} for write", pair_key_write.first.atom_key());
-                            key_segs->emplace_back(std::move(pair_key_write.first));
-                        }
-                    }));
```

Comment:
Below is definitely over-wrapped. We need longer lines and maybe don't enable strings to be auto-split